### ISSUE TITLE
[BUG] one_hot_encoding is slower due to iteratively creating columns

### ISSUE THREAD

**Issue by galipremsagar** [rapidsai/cudf#2246]
**Describe the bug**
For a dataset of size 20000 records & spanning across 10 partitions in a dask dataframe, it takes about 5 seconds for `get_dummies` to complete it's execution. After looking at the profile we find that `get_dummies` internally calls `one_hot_encoding` which is taking most of the time needed for execution. This is because we are iteratively generating a series for every category of the column. 

![image](https://user-images.githubusercontent.com/11664259/61250898-8ee8f500-a71e-11e9-8a73-c51c4f8700cb.png)
https://github.com/rapidsai/cudf/blob/branch-0.9/python/cudf/cudf/dataframe/dataframe.py#L1572

The problem is the same with `Series.one_hot_encoding` too: https://github.com/rapidsai/cudf/blob/branch-0.9/python/cudf/cudf/dataframe/series.py#L1578
![image](https://user-images.githubusercontent.com/11664259/61250952-c192ed80-a71e-11e9-8566-8c48e253460e.png)

**Steps/Code to reproduce bug**

```python
from dask_cuda import LocalCUDACluster
from dask.distributed import Client
gdf = cudf.datasets.randomdata(nrows=20000, dtypes={"C": int,"first": 'category',"b": float,"second": 'category'})
df = gdf.to_pandas()
ddf = dd.from_pandas(df, npartitions=10)
gddf = dask_cudf.from_cudf(gdf, npartitions=10)


In [10]: %time x = dd.get_dummies(gddf).compute()
CPU times: user 4.38 s, sys: 184 ms, total: 4.56 s
Wall time: 4.56 s

In [11]: %time x = dd.get_dummies(ddf).compute()
CPU times: user 60 ms, sys: 12 ms, total: 72 ms
Wall time: 68 ms
```

**Expected behavior**
We should ideally be moving `Series.one_hot_encoding` to libcudf to solve this? 

**Environment overview (please complete the following information)**
 - Environment location: Docker
 - Method of cuDF install: from source

**Environment details**
`cudf/print_env.sh`: 
[env.txt](https://github.com/rapidsai/cudf/files/3394320/env.txt)

**Additional context**
since `get_dummies` uses `one_hot_encoding` internally, this performance issue hits get_dummies as well.

**Comment by jrhemstad** [comment]
> We should ideally be moving Series.one_hot_encoding to libcudf to solve this?

Yep, pretty much.

**Comment by jrhemstad** [comment]
That said, I believe @kkraus14 has mentioned in the past that there's low hanging fruit in the code for creating columns where it's doing an unnecessary element-wise equality check.

Ultimately we would like to move all of this functionality into libcudf, but I think there's some easy fixes on the Python side in the meantime.

**Comment by kkraus14** [comment]
@jrhemstad this one is problematic, because we need to create a new binary encoded column for each value in the categories for each column getting encoded. We're doing this currently by essentially iterating over the categories (which are on the device which doesn't help us here) serially and running an `eq` op to get the binary encoded column.

**Comment by jrhemstad** [comment]
> @jrhemstad this one is problematic, because we need to create a new binary encoded column for each value in the categories for each column getting encoded. We're doing this currently by essentially iterating over the categories (which are on the device which doesn't help us here) serially and running an `eq` op to get the binary encoded column.

That sounds fairly reasonable. I wouldn't expect that to take ~5seconds, unless there was like 10,000 unique values in the categories.

**Comment by kkraus14** [comment]
Fixed by #2296
